core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/salt/rotatepassword/RotateSaltPasswordActions.java

```diff
@@ -69,9 +69,19 @@ protected Selectable createRequest(RotateSaltPasswordContext context) {
             @Override
             protected void doExecute(RotateSaltPasswordContext context, RotateSaltPasswordSuccessResponse payload, Map<Object, Object> variables)
                     throws Exception {
-                LOGGER.info("Rotating salt password for stack {} finished", context.getStack().getResourceCrn());
-                stackUpdaterService.updateStatus(payload.getResourceId(), DetailedStackStatus.AVAILABLE,
-                        CLUSTER_SALT_PASSWORD_ROTATE_FINISHED, "SaltStack user password rotated");
+                LOGGER.info("Rotating salt password for stack {} finished, restoring previous status: {}",
+                        context.getStack().getResourceCrn(), context.getPreviousStackStatus());
+                String statusReason = "SaltStack user password rotated, restored previous status";
+                String previousStatusReason = context.getPreviousStackStatus().getStatusReason();
+                if (!previousStatusReason.isBlank()) {
+                    if (previousStatusReason.startsWith(statusReason)) {
+                        statusReason = previousStatusReason;
+                    } else {
+                        statusReason += ": " + previousStatusReason;
+                    }
+                }
```

Comment:
this would be a good class candidate in common, so it won't have to be repeated in freeipa module